### PR TITLE
Add element_type member of TaggedVector

### PR DESCRIPTION
**Pull request by EmilyBourne**

### CODE REVIEW

File: include/ddc/detail/tagged_vector.hpp [hunk 1]
```diff
@@ -242,6 +242,8 @@ class TaggedVector : public ConversionOperators<TaggedVector<ElementType, Tags..
     std::array<ElementType, sizeof...(Tags)> m_values;
 
 public:
+    using element_type = ElementType;
```
Comment (tpadioleau) [inline include/ddc/detail/tagged_vector.hpp:245]: To mimic `std::array` I would expect `value_type`, what do you think ?
```suggestion
    using value_type = ElementType;
```

Comment (EmilyBourne) [reply include/ddc/detail/tagged_vector.hpp:245]: This is fine for me. I used `element_type` to match `Chunk`

Comment (EmilyBourne) [reply include/ddc/detail/tagged_vector.hpp:245]: It seems that there is a `value_type` too. What is the difference?

**Review by tpadioleau — APPROVED** [review]

File: include/ddc/detail/tagged_vector.hpp (hunk 1 above)
Comment (tpadioleau) [reply include/ddc/detail/tagged_vector.hpp]: In `ChunkSpan` ? It is inspired from `mdspan`, they seem to define it as `using value_type = remove_cv_t<element_type>`

**Pull request merged**